src/fable/Fable.Compiler/Replacements.fs

```diff
@@ -400,17 +400,10 @@ module private AstPass =
                     | Some t -> Some(Fable.Array t)
                     | None -> None
                 | Fable.Tuple ts ->
-                    (ts, (false, []))
-                    ||> List.foldBack (fun t (inflate, acc) ->
-                        match needsInflate genArgsMap t with
-                        | Some t -> true, t::acc
-                        | None -> inflate, t::acc)
-                    |> function
-                    | true, ts -> Some(Fable.Tuple ts)
-                    | false, _ -> None
+                    Some(Fable.Tuple ts)
                 | Fable.DeclaredType(ent, [genArg])
                     when ent.FullName = "Microsoft.FSharp.Core.FSharpOption" ->
-                    needsInflate genArgsMap genArg
+                    Some(Fable.DeclaredType(ent, [genArg]))
```

Comment:
This and the removal of code for tuples above prevents optimization of the schema (for example, a string option or an int*int tuple don't need any special JSON deserialization treatment so they don't need to be included in the schema).